src/rust/ensogl/lib/core/src/animation/physics/inertia.rs

```diff
@@ -220,14 +220,19 @@ impl Thresholds {
 /// ```
 #[derive(Clone,Copy,Debug,Default)]
 pub struct SimulationData<T> {
-    value        : T,
-    target_value : T,
-    velocity     : T,
-    mass         : Mass,
-    spring       : Spring,
-    drag         : Drag,
-    thresholds   : Thresholds,
-    active       : bool,
+    // We store the current value as an offset from the target rather than an absolute value,
+    // because this provides higher precision when the value is getting close to the target and `T`
+    // is a type of floating-point numbers, vectors of floating point numbers, or similar. The main
+    // benefit is that we avoid non-termination that could otherwise happen due to rounding errors
+    // when the target and current value are large.
```

Comment:
I do not understand this description. "this provides higher precision when the value is getting close to the target" - why? Could you describe it with some examples and better explanation, please?

Reply:
I have tried to improve the explanation. Let me know if it is better.